tests/unit/s2n_resume_test.c

```diff
@@ -230,9 +216,27 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
             EXPECT_SUCCESS(s2n_realloc(&conn->client_ticket, client_ticket_size));
 
-            uint8_t versions[] = { S2N_TLS12, S2N_TLS13 };
-            for (size_t i = 0; i < s2n_array_len(versions); i++) {
-                conn->actual_protocol_version = versions[i];
+            /* TLS 1.2 */
+            {
+                conn->actual_protocol_version = S2N_TLS12;
+                int session_length = s2n_connection_get_session_length(conn);
+                EXPECT_NOT_EQUAL(session_length, 0);
+
+                /* Result matches size expected by s2n_connection_get_session */
+                DEFER_CLEANUP(struct s2n_blob session_data = { 0 }, s2n_free);
+                EXPECT_SUCCESS(s2n_alloc(&session_data, session_length));
+                EXPECT_SUCCESS(s2n_connection_get_session(conn, session_data.data, session_data.size));
+            }
+
+            /* TLS 1.3 */
+            {
+                conn->actual_protocol_version = S2N_TLS13;
+                conn->secure->cipher_suite = &s2n_tls13_aes_256_gcm_sha384;
```

Comment:
Just give each test their own connection. Sharing mutable state between tests is risky.